### PR TITLE
Add/Edit books fully functional and update firestore accordingly

### PR DESCRIPTION
**Pull request by KateVlaar**
BookListAdapter is now a FirestoreRecyclerAdapter which updates the RecyclerView in real time with Firestore, its pretty smooth. Because of this RecyclerViewTouchListener is no longer needed but I left it in just in case any pending PR's are using it. No visual changes.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="false" project-jdk-name="1.8" project-jdk-type="JavaSDK" />
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="1.8" project-jdk-type="JavaSDK" />
```
Comment (naviza) [inline .idea/misc.xml:3]: we're keeping this false now? I think I saw a previous merged PR that sets this to false

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java [hunk 2]
```diff
@@ -36,6 +38,26 @@ protected void onCreate(Bundle savedInstanceState)
         authorView = findViewById(R.id.author_edit_text);
         descriptionView = findViewById(R.id.description_edit_text);
         isbnView = findViewById(R.id.isbn_edit_text);
+
+        // If editing a book, prepopulate text fields with their old values
+        int requestCode = -1;
+        if (getIntent().getSerializableExtra("requestCode") != null)
```
Comment (naviza) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java:44]: For using integers with intent, I think best practice is to use [putExtra()](https://stackoverflow.com/questions/7074097/how-to-pass-integer-from-one-activity-to-another) in the parent activity and using getIntExtra() for the child activity. Using getSerializable(...) is usually used for passing in user defined objects around activities.

**Review by naviza — CHANGES_REQUESTED** [review]
Haven't fully tested it yet but I have a few small nits

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java (hunk 2 above)
Comment (KateVlaar) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java:44]: Nice catch, fixed

File: .idea/misc.xml (hunk 1 above)
Comment (huddy987) [reply .idea/misc.xml:3]: Keep this as true. We originally left it false but everyone's commits are doing the same thing so obviously true is the right choice.

**Review by huddy987 — APPROVED** [review]

**Review by naviza — APPROVED** [review]

**Pull request merged**